### PR TITLE
Add the entity class for xDS flow control functionality

### LINKED ISSUE #1515: Supports xDS protocol to communicate with the Control Plane of Service Mesh
# Value
Sermant is a Proxyless Service Mesh Solution with better performance.The support for the xDS protocol enables Sermant to better integrate into the Cloud Native Ecosystem. In Istio, Sermant can replace Envoy as the data plane to obtain directly the configuration of the control plane, and avoid performance loss and architecture complexity caused by Sidecar containers.
# Objective
Sermant can connect to the control plane of Istio through the xDS API interface to obtain the configurations of CDS, EDS, LDS, and RDS, and parse configurations to data model of the Sermant. Sermant will  implement gradually capabilities such as service discovery , route configuration and flow control, and the the host microservice can implements the Proxyless Mesh capability by mounting Sermant to  invoke directly the upstream service registered with the control plane, such as istiod.
# RoadMap
**v2.0.0 (2024/06/30)**
Supports xDS protocol for service discovery
**v2.1.0 (2024/09/30)**
Supports xDS protocol for routing and load balancing
**v2.2.0 (2024/12/30)**
Supports xDS protocol for flow control

### PR DESCRIPTION
**Pull request by hanbingleixue**
**What type of PR is this?**

Feature

**What this PR does / why we need it?**

Add the entity class for Xds flow control functionality

**Which issue(s) this PR fixes？**

Fixes #1515

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsHeaderOption.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+public class XdsHeaderOption {
+    /**
+     * Header name/ value pair that this option applies to
+     */
+    private XdsHeader header;
+
+    /**
+     * Should the value be appended
+     */
+    private boolean append;
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsHeaderOption.java]: enabledAppend

Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsHeaderOption.java]: Resolved

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsDelay.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * Xds Delay information
+ *
+ * @author zhp
+ * @since 2024-11-18
+ */
+public class XdsDelay {
+    /**
+     * the specified delay will be injected before a new request
+     */
+    private long fixedDelay;
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsDelay.java]: specify unit of fixedDelay, s or ms

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsFractionalPercent.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+package io.sermant.core.service.xds.entity;
+
+/**
+ * Xds FractionalPercent configuration information
+ *
+ * @author zhp
+ * @since 2024-11-18
+ */
+public class XdsFractionalPercent {
+    /**
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsFractionalPercent.java]: Is there any relation with xDS for this entity？ If not, modify the name

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsOutlierDetection.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+     * The number of consecutive gateway failures (502, 503, 504 status codes) before a consecutive gateway failure
+     * ejection occurs
+     */
+    private int consecutiveGatewayFailure;
+
+    /**
+     * The number of consecutive 5xx responses or local origin errors that are mapped to 5xx error codes before a
+     * consecutive 5xx ejection occurs
+     */
+    private int consecutive5Xx;
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsOutlierDetection.java]: 5Xx seems not right

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsDelay.java (hunk 2 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsDelay.java]: Resolved

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsFractionalPercent.java (hunk 3 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsFractionalPercent.java]: Resolved

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsOutlierDetection.java (hunk 4 above)
Comment (hanbingleixue) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/service/xds/entity/XdsOutlierDetection.java]: Resolved

**Review by lilai23 — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**